typescript/src/ai-sdk/essentials.ts

```diff
@@ -7,37 +7,61 @@ import {
 import {contextToTools} from '../shared/tools';
 import type {Configuration} from '../types/configuration';
 import CommercetoolsTool from './tool';
+import {AuthConfig} from '../types/auth';
+
+type DeprecatedConstructorParams = {
+  clientId: string;
+  clientSecret: string;
+  authUrl: string;
+  projectKey: string;
+  apiUrl: string;
+  configuration: Configuration;
+};
+
+type NewConstructorParams = {
+  authConfig: AuthConfig;
+  configuration: Configuration;
+};
+
 class CommercetoolsAgentEssentials {
   private _commercetools: CommercetoolsAPI;
 
   tools: {[key: string]: Tool};
 
+  /**
+   * @deprecated Use the new constructor with authConfig instead
+   */
   constructor({
     clientId,
     clientSecret,
     authUrl,
     projectKey,
     apiUrl,
     configuration,
-  }: {
-    clientId: string;
-    clientSecret: string;
-    authUrl: string;
-    projectKey: string;
-    apiUrl: string;
-    configuration: Configuration;
-  }) {
+  }: DeprecatedConstructorParams);
```

Comment:
To avoid breaking changes, we've introduced a new constructor and kept the old one.
I've removed usages of the old constructor from the docs. We can remove it later.